sdk/src/types/block/payload/signed_transaction/mod.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module describing the transaction payload.
+
+mod transaction;
+mod transaction_id;
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use packable::{error::UnpackError, packer::Packer, unpacker::Unpacker, Packable, PackableExt};
+
+pub(crate) use self::transaction::{ContextInputCount, InputCount, OutputCount};
+pub use self::{
+    transaction::{Transaction, TransactionBuilder, TransactionCapabilities, TransactionCapabilityFlag},
+    transaction_id::{TransactionHash, TransactionId},
+};
+use crate::types::block::{protocol::ProtocolParameters, unlock::Unlocks, Error};
+
+/// A signed transaction to move funds.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SignedTransactionPayload {
+    transaction: Transaction,
+    unlocks: Unlocks,
+}
+
+impl SignedTransactionPayload {
+    /// The payload kind of a [`SignedTransactionPayload`].
+    pub const KIND: u8 = 1;
+
+    /// Creates a new [`SignedTransactionPayload`].
+    pub fn new(transaction: Transaction, unlocks: Unlocks) -> Result<Self, Error> {
+        verify_transaction_unlocks(&transaction, &unlocks)?;
+
+        Ok(Self { transaction, unlocks })
+    }
+
+    /// Return the transaction of a [`SignedTransactionPayload`].
+    pub fn transaction(&self) -> &Transaction {
+        &self.transaction
+    }
+
+    /// Return unlocks of a [`SignedTransactionPayload`].
+    pub fn unlocks(&self) -> &Unlocks {
+        &self.unlocks
+    }
+
+    /// Computes the identifier of a [`SignedTransactionPayload`].
+    pub fn id(&self) -> TransactionId {
+        let mut hasher = Blake2b256::new();
+
+        hasher.update(Self::KIND.to_le_bytes());
+        hasher.update(self.pack_to_vec());
+
+        TransactionHash::new(hasher.finalize().into()).into_transaction_id(self.transaction.creation_slot())
+    }
+}
+
+impl Packable for SignedTransactionPayload {
```

Comment:
Can be derived?

Reply:
Looks like it

Reply:
Ah no, it can't. The verify method needs both fields :/

Reply:
Would be nice to have a post unpack verify_with or something

Reply:
Ah, ok